packages/react-vapor/src/components/textInput/TextInputContext.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+
+type TextInputAction =
+    | {type: 'change-value'; payload: string}
+    | {type: 'show-validation'}
+    | {type: 'hide-validation'}
+    | {type: 'set-valid'}
+    | {type: 'set-invalid'}
+    | {type: 'set-warning'}
+    | {type: 'set-message'; payload: string};
+
+type TextInputState = {
+    value: string;
+    status: 'valid' | 'invalid' | 'warning' | 'indeterminate';
+    visibleStatus: boolean;
+    message: string;
+};
+const defaultState: TextInputState = {
+    value: '',
+    status: 'indeterminate',
+    visibleStatus: false,
+    message: '',
+};
+
+type AllTextInputsAction = {type: string; payload: TextInputAction};
+type AllTextInputsState = Record<string, TextInputState>;
+
+const context = React.createContext<{state: AllTextInputsState; dispatch: React.Dispatch<AllTextInputsAction>}>(
+    undefined
+);
+
+const oneInputReducer = (state: TextInputState, action: TextInputAction): TextInputState => {
+    switch (action.type) {
+        case 'change-value':
+            return action.payload === state.value ? state : {...state, value: action.payload};
+        case 'show-validation':
+            return {...state, visibleStatus: true};
+        case 'hide-validation':
+            return {...state, visibleStatus: false};
+        case 'set-valid':
+            return {...state, status: 'valid'};
+        case 'set-invalid':
+            return {...state, status: 'invalid'};
+        case 'set-warning':
+            return {...state, status: 'warning'};
+        case 'set-message':
+            return action.payload === state.message ? state : {...state, message: action.payload};
+        default:
+            throw new Error(`Unhandled action type: ${(action as {type: 'string'}).type}.`);
+    }
+};
+
+const allInputsReducer = (state: AllTextInputsState, action: AllTextInputsAction): AllTextInputsState => {
+    const oldState: TextInputState = state[action.type] || defaultState;
+    const newState: TextInputState = oneInputReducer(oldState, action.payload);
+    return oldState === newState
+        ? state
+        : {
+              ...state,
+              [action.type]: newState,
+          };
+};
+
+export const TextInputProvider: React.FunctionComponent = ({children}) => {
```

Comment:
As discussed in Slack, I find the `TextInputProvider` a bit too specific, I'd instead target something like `FormProvider` that exposes a context that can support a couple of different inputs.

Reply:
Implemented the `FormProvider`, it will support multiple different form components.